### PR TITLE
feat(core): dynamic switch view of calendar when value changes

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

https://user-images.githubusercontent.com/12021443/209865177-cd4d4d3f-830a-4c35-bf26-206374db9ef0.mov

## What is the new behavior?

https://user-images.githubusercontent.com/12021443/209865462-0522aed3-4928-4fc3-8629-3bcace8cef8f.mov

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/core/components/calendar/calendar.component.ts [hunk 1]
```diff
@@ -136,4 +148,13 @@ export class TuiCalendarComponent implements TuiWithOptionalMinMax<TuiDay> {
         this.hoveredItem = day;
         this.hoveredItemChange.emit(day);
     }
+
+    private getTuiMonth(value: TuiDay | TuiDayRange | null): TuiMonth | null {
```
Comment (waterplea) [inline projects/core/components/calendar/calendar.component.ts:152]: Tui is unnecessary

File: projects/core/components/calendar/calendar.component.ts [hunk 2]
```diff
@@ -136,4 +148,13 @@ export class TuiCalendarComponent implements TuiWithOptionalMinMax<TuiDay> {
...
+        const possibleTuiDay = value as TuiDay;
```
Comment (waterplea) [inline projects/core/components/calendar/calendar.component.ts:153]: You can do `instanceof` test since `TuiDay` is a class.

Comment (splincode) [reply projects/core/components/calendar/calendar.component.ts]: fixed

File: projects/core/components/calendar/calendar.component.ts (hunk 1 above)
Comment (splincode) [reply projects/core/components/calendar/calendar.component.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Review by vladimirpotekhin — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Pull request merged**